core/common/src/main/java/alluxio/underfs/BaseUnderFileSystem.java

```diff
@@ -82,6 +86,35 @@ public boolean deleteDirectory(String path) throws IOException {
     return deleteDirectory(path, DeleteOptions.defaults());
   }
 
+  @Override
+  public OutputStream createNonexistingFile(String path) throws IOException {
+    return retryOnException(() -> create(path, CreateOptions.defaults(mAlluxioConf)),
```

Comment:
what do you think of moving the default retrying implementations to `ObjectUnderFileSystem`? Then we don't need to override existing implementations in `ConsistentUnderFileSystem`. If we add a new retry-style method to `UnderFileSystem`, it would be nice if the compiler forced us to add a consistent implementation to `ConsistentUnderFileSystem`. Right now, we could easily forget to override a retry-style method in `ConsistentUnderFileSystem`, and end up doing retries there.

Reply:
Agree! by now the eventual consistency issue only occur in the Object storages. if in the future, other kind of storages also have this issue, we can revert to implement retry in `BaseUnderFileSystem` and overwrite in `ConsistentUnderFileSystem`.

Reply:
My general point is that I don't like having default implementations in `BaseFileSystem`. That way, if we add a new method in `BaseFileSystem`, we are forced by the compiler to provide appropriate implementations in both `ConsistentUnderFileSystem` and `ObjectUnderFileSystem`. Right now, if we add a new method to `BaseUnderFileSystem`, we also need to remember to override it in `ObjectUnderFileSystem`. If we forget, `ObjectUnderFileSystem` will use the default behavior, and we could have consistency issues again.

Reply:
Oh, I see. Do not give default retry in `BaseFileSystem` to force us implement `consistent ways` in `ConsistentUnderFileSystem` and deal with eventual consistency in `objectUnderFileSystem`. Thanks!